File: uva_11565.cpp
# include <iostream>
using namespace std;
int main(void){
    int num;
    cin >> num;
    while(num--){
        int A, B, C;
        bool check=true;
        cin >> A >> B >> C;
        for (int x=-100; 3*x<A && check; ++x){
            if (!x)
                continue;
            if (x*x>C || B%x)
                continue;
            for (int y=x+1; x+2*y<A && check; ++y){
                int z=A-x-y;
                if (y < z && x*y*z == B && x*x+y*y+z*z == C){
                    check=false;
                    cout << x << ' ' << y << ' ' << z << endl;
                }
            }
        }
        if (check)
            cout << "No solution." << endl;
    }
    return 0;
}

File: uva_11565.py
num=int(input())
while num:
    num-=1
    check=True
    A, B, C=input().split()
    A, B, C=int(A), int(B), int(C)
    x=-101
    while 3*x<A and check:
        x+=1
        if not x:
            continue
        if x*x>C or B%x:
            continue
        y=x
        while x+2*y<A and check:
            y+=1
            z=A-x-y
            if y < z and x*y*z == B and x*x+y*y+z*z == C:
                check=False
                print('{} {} {}'.format(x, y, z))
    if check:
        print('No solution.')